src/main/scala/org/deepdive/ddlog/DeepDiveLogParser.scala

```diff
@@ -329,10 +328,10 @@ class DeepDiveLogParser extends JavaTokenParsers {
   ( implyHeadAtoms ^^ {
       InferenceRuleHead(FactorFunction.Imply, _)
     }
-  | "@semantics(linear)" ~> implyHeadAtoms ^^ {
+  | ("@semantics" ~ "(" ~ "\"linear\"" ~ ")") ~> implyHeadAtoms ^^ {
       InferenceRuleHead(FactorFunction.Linear, _)
     }
-  | "@semantics(ratio)" ~> implyHeadAtoms ^^ {
+  | ("@semantics" ~ "(" ~ "\"ratio\"" ~ ")") ~> implyHeadAtoms ^^ {
```

Comment:
Also, please add parse error test cases for using bad values for `@semantics`, `@mode`, etc.